Test does not work with Node v22

The current vitest works only with Node v20 (previous LTS), but not with v22 (current LTS).

There seem to be two issues:
- better-sqlite3 error (on Linux and macOS)
- "no test files found" error and internal error with 500 (reported by @MathurAditya724 on Windows machine)

## Node v20

```shell
$ node -v
v20.18.2

$ pnpm --filter @tsky/client test

> @tsky/client@0.0.1 test /tmp/tsky/packages/client
> vitest


 DEV  v3.0.5 /tmp/tsky/packages/client

🌐 Created test network:
- pds: http://localhost:44389
- plc: http://localhost:45513
🙋 Created new account: @alice.test
🙋 Created new account: @bob.test
 ✓ src/tsky/tsky.test.ts (2 tests) 738ms
   ✓ createAgent > can create agent for Alice 381ms
   ✓ createAgent > can resume from stored session 355ms
 ✓ src/user/profile.test.ts (2 tests) 1989ms
   ✓ profile > Getting alice's profile 1248ms
   ✓ profile > Getting bob's profile 741ms
 ✓ src/user/preferences/preferences.test.ts (2 tests) 2269ms
   ✓ preferences > .get() 1259ms
   ✓ preferences > .set() 1010ms
 ✓ src/feed/feed.test.ts (1 test) 2573ms
   ✓ feed > .getFeed() 2572ms

 Test Files  4 passed (4)
      Tests  7 passed (7)
   Start at  16:01:16
   Duration  3.77s (transform 100ms, setup 0ms, collect 281ms, tests 7.57s, environment 0ms, prepare 152ms)

 PASS  Waiting for file changes...
       press h to show help, press q to quit
```


## Node v22

### better-sqlite3 error

```shell
$ node -v
v22.12.0

$ pnpm --filter @tsky/client test

> @tsky/client@0.0.1 test /tmp/tsky/packages/client
> vitest


 DEV  v3.0.5 /tmp/tsky/packages/client

No test files found. You can change the file name pattern by pressing "p"

include: **/*.{test,spec}.?(c|m)[jt]s?(x)
exclude:  **/node_modules/**, **/dist/**, **/cypress/**, **/.{idea,git,cache,output,temp}/**, **/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build,eslint,prettier}.config.*


⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error: The module '/tmp/tsky/node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3/build/Release/better_sqlite3.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 115. This version of Node.js requires
NODE_MODULE_VERSION 127. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
 ❯ Object..node node:internal/modules/cjs/loader:1734:18
 ❯ Module.load node:internal/modules/cjs/loader:1318:32
 ❯ Function._load node:internal/modules/cjs/loader:1128:12
 ❯ TracingChannel.traceSync node:diagnostics_channel:322:14
 ❯ wrapModuleLoad node:internal/modules/cjs/loader:219:24
 ❯ Module.require node:internal/modules/cjs/loader:1340:12
 ❯ require node:internal/modules/helpers:138:16
 ❯ bindings ../../node_modules/.pnpm/bindings@1.5.0/node_modules/bindings/bindings.js:112:48
 ❯ new Database ../../node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3/lib/database.js:48:64
 ❯ Database.sqlite ../../node_modules/.pnpm/@atproto+pds@0.4.94/node_modules/@atproto/pds/dist/db/db.js:37:26

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: { code: 'ERR_DLOPEN_FAILED' }
```

By manually rebuilding better-sqlite3, I was able to run vitest on Node v22 but this shouldn't be done manually.



<details>
<pre>
$ cd /tmp/tsky/node_modules/.pnpm/better-sqlite3@10.1.0/
$ pnpm run build-release

> better-sqlite3@10.1.0 build-release /tmp/tsky/node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3
> node-gyp rebuild --release

gyp info it worked if it ends with ok
gyp info using node-gyp@11.0.0
gyp info using node@22.12.0 | linux | x64
gyp info find Python using Python version 3.13.2 found at "/home/shuuji3/.local/share/mise/installs/python/3.13.2/bin/python3"

gyp http GET https://nodejs.org/download/release/v22.12.0/node-v22.12.0-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v22.12.0/node-v22.12.0-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v22.12.0/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v22.12.0/SHASUMS256.txt
gyp info spawn /home/shuuji3/.local/share/mise/installs/python/3.13.2/bin/python3
gyp info spawn args [
gyp info spawn args '/home/shuuji3/.local/share/pnpm/global/5/.pnpm/node-gyp@11.0.0/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args 'binding.gyp',
gyp info spawn args '-f',
gyp info spawn args 'make',
gyp info spawn args '-I',
gyp info spawn args '/tmp/tsky/node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3/build/config.gypi',
gyp info spawn args '-I',
gyp info spawn args '/home/shuuji3/.local/share/pnpm/global/5/.pnpm/node-gyp@11.0.0/node_modules/node-gyp/addon.gypi',
gyp info spawn args '-I',
gyp info spawn args '/home/shuuji3/.cache/node-gyp/22.12.0/include/node/common.gypi',
gyp info spawn args '-Dlibrary=shared_library',
gyp info spawn args '-Dvisibility=default',
gyp info spawn args '-Dnode_root_dir=/home/shuuji3/.cache/node-gyp/22.12.0',
gyp info spawn args '-Dnode_gyp_dir=/home/shuuji3/.local/share/pnpm/global/5/.pnpm/node-gyp@11.0.0/node_modules/node-gyp',
gyp info spawn args '-Dnode_lib_file=/home/shuuji3/.cache/node-gyp/22.12.0/<(target_arch)/node.lib',
gyp info spawn args '-Dmodule_root_dir=/tmp/tsky/node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3',
gyp info spawn args '-Dnode_engine=v8',
gyp info spawn args '--depth=.',
gyp info spawn args '--no-parallel',
gyp info spawn args '--generator-output',
gyp info spawn args 'build',
gyp info spawn args '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/tmp/tsky/node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3/build'
  TOUCH ba23eeee118cd63e16015df367567cb043fed872.intermediate
  ACTION deps_sqlite3_gyp_locate_sqlite3_target_copy_builtin_sqlite3 ba23eeee118cd63e16015df367567cb043fed872.intermediate
  TOUCH Release/obj.target/deps/locate_sqlite3.stamp
  CC(target) Release/obj.target/sqlite3/gen/sqlite3/sqlite3.o
rm -f Release/obj.target/deps/sqlite3.a Release/obj.target/deps/sqlite3.a.ar-file-list; mkdir -p `dirname Release/obj.target/deps/sqlite3.a`
ar crs Release/obj.target/deps/sqlite3.a @Release/obj.target/deps/sqlite3.a.ar-file-list
  COPY Release/sqlite3.a
  CXX(target) Release/obj.target/better_sqlite3/src/better_sqlite3.o
In file included from ./src/better_sqlite3.lzz:11,
                 from ../src/better_sqlite3.cpp:4:
/home/shuuji3/.cache/node-gyp/22.12.0/include/node/node.h:1246:7: warning: cast between incompatible function types from ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>)’ to ‘node::addon_context_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, void*)’} [-Wcast-function-type]
 1246 |       (node::addon_context_register_func) (regfunc),                  \
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shuuji3/.cache/node-gyp/22.12.0/include/node/node.h:1264:3: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE_X’
 1264 |   NODE_MODULE_CONTEXT_AWARE_X(modname, regfunc, NULL, 0)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/shuuji3/.cache/node-gyp/22.12.0/include/node/node.h:1295:3: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE’
 1295 |   NODE_MODULE_CONTEXT_AWARE(NODE_GYP_MODULE_NAME,                     \
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
./src/better_sqlite3.lzz:67:1: note: in expansion of macro ‘NODE_MODULE_INIT’
./src/objects/database.lzz: In static member function ‘static void Database::JS_new(const v8::FunctionCallbackInfo<v8::Value>&)’:
./src/objects/database.lzz:180:21: warning: variable ‘status’ set but not used [-Wunused-but-set-variable]
./src/util/custom-table.lzz: At global scope:
./src/util/custom-table.lzz:45:9: warning: missing initializer for member ‘sqlite3_module::xIntegrity’ [-Wmissing-field-initializers]
./src/util/custom-table.lzz:72:9: warning: missing initializer for member ‘sqlite3_module::xIntegrity’ [-Wmissing-field-initializers]
./src/util/data.lzz: In function ‘v8::Local<v8::Value> Data::GetValueJS(v8::Isolate*, sqlite3_stmt*, int, bool)’:
./src/util/data.lzz:73:92: warning: this statement may fall through [-Wimplicit-fallthrough=]
./src/util/data.lzz:73:197: note: here
./src/util/data.lzz: In function ‘v8::Local<v8::Value> Data::GetValueJS(v8::Isolate*, sqlite3_value*, bool)’:
./src/util/data.lzz:77:81: warning: this statement may fall through [-Wimplicit-fallthrough=]
./src/util/data.lzz:77:175: note: here
  SOLINK_MODULE(target) Release/obj.target/better_sqlite3.node
  COPY Release/better_sqlite3.node
  CC(target) Release/obj.target/test_extension/deps/test_extension.o
  SOLINK_MODULE(target) Release/obj.target/test_extension.node
  COPY Release/test_extension.node
rm ba23eeee118cd63e16015df367567cb043fed872.intermediate
make: Leaving directory '/tmp/tsky/node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3/build'
gyp info ok 

$ pnpm --filter @tsky/client test

> @tsky/client@0.0.1 test /tmp/tsky/packages/client
> vitest


 DEV  v3.0.5 /tmp/tsky/packages/client

🌐 Created test network:
- pds: http://localhost:40263
- plc: http://localhost:42477
🙋 Created new account: @alice.test
🙋 Created new account: @bob.test
 ✓ src/tsky/tsky.test.ts (2 tests) 734ms
   ✓ createAgent > can create agent for Alice 378ms
   ✓ createAgent > can resume from stored session 356ms
 ❯ src/user/profile.test.ts (2 tests | 2 failed) 940ms
   × profile > Getting alice's profile 687ms
     → agent.user.profile is not a function
   × profile > Getting bob's profile 252ms
     → agent.user.profile is not a function
 ✓ src/user/preferences/preferences.test.ts (2 tests) 2296ms
   ✓ preferences > .get() 1247ms
   ✓ preferences > .set() 1049ms
 ✓ src/feed/feed.test.ts (1 test) 2424ms
   ✓ feed > .getFeed() 2423ms
...
</pre>
</details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -23,7 +23,7 @@ jobs:
       - uses: pnpm/action-setup@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: 20
+          node-version: 22
           cache: pnpm
 
       - name: 📦 Install dependencies

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -19,7 +19,7 @@ jobs:
       - name: Set node
         uses: actions/setup-node@v4
         with:
-          node-version: 18
+          node-version: 22
 
       - run: npx changelogithub
         env:

--- a/.github/workflows/update-lexicons.yml
+++ b/.github/workflows/update-lexicons.yml
@@ -24,7 +24,7 @@ jobs:
       - uses: pnpm/action-setup@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: 20
+          node-version: 22
           cache: pnpm
 
       - name: 📦 Install dependencies

